File: CMemoryTest/memorytest.c
#include <stdio.h>
struct Example{
    int a_integer;
    char a_string[3];
};

int main (){
  struct Example someData;
  struct Example someOtherData;
  
  someData.a_integer = 1;
  someData.a_string[0] = 'h';
  someData.a_string[1] = 'i';

  someOtherData.a_integer = 2;
  someOtherData.a_string[0] = 'B';
  someOtherData.a_string[1] = 'y';
  someOtherData.a_string[2] = 'e';

  someData = someOtherData;

  printf("%c %c %c", someData.a_string[0], someData.a_string[1], someData.a_string[2]);
  printf("%i", someData.a_integer);

};

File: RustMemoryTest/memorytest.rs
struct Example{
    a_integer: i32,
    a_string: String, 
}
impl Example{
    fn toString(&self) -> String{
        format!("{} - {}", self.a_integer, self.a_string)
    }
    fn println(&self){
        println!("{}",self.toString());
    }
}

fn main(){

    let mut someData = Example{
        a_integer: 1,
        a_string: String::from("hi"),
    };
    let someOtherData = Example{
        a_integer: 2,
        a_string: String::from("bye"),
    }; 
    someData.println();
    someData = someOtherData;
    someData.println();
    someOtherData.println();
    let mut SampleArray: [i32; 5] = [0; 5];
    for n in 0..SampleArray.len() {
        SampleArray[n] = 5;
    }
}